app/lib/actions/katello/content_view/promote_to_environment.rb

```diff
@@ -41,7 +41,7 @@ def humanized_name
 
         def run
           environment = ::Katello::KTEnvironment.find(input[:environment_id])
-          if ::Katello::CapsuleContent.sync_needed?(environment)
+          if ::Katello::SmartProxy.sync_needed?(environment)
```

Comment:
I think you mean `::SmartProxy` here, `::Katello::SmartProxy`` doesn't seem to exist.